Raven.Database/Indexing/IndexStorage.cs

```diff
@@ -253,8 +253,23 @@ private Index OpenIndex(string indexName, bool onStartup, bool forceFullIndexChe
                         throw new InvalidOperationException("Could not open / create index" + indexName + ", reset already tried", e);
 
                     if (indexImplementation != null)
+                    {
                         indexImplementation.Dispose();
 
+                        // index disposal just closed the lucene directory, let's try to reopen it
+                        try
+                        {
+                            luceneDirectory = OpenOrCreateLuceneDirectory(indexDefinition, createIfMissing: resetTried, forceFullExistingIndexCheck: forceFullIndexCheck);
+                        }
+                        catch (Exception ex)
```

Comment:
What will cause this error here?

Reply:
It's very unlikely we will get an error here as we already succeeded in opening it once, so I don't see why we could fail on the second attempt. Just defensive coding.

Reply:
No, the user issue  was that we succeeded in opening the lucene directory and creating an index instance:

https://github.com/ravendb/ravendb/blob/8936425e9eda9975a189890cae73d16eaba78b85/Raven.Database/Indexing/IndexStorage.cs#L201-L202

but failed in:

https://github.com/ravendb/ravendb/blob/8936425e9eda9975a189890cae73d16eaba78b85/Raven.Database/Indexing/IndexStorage.cs#L204

In result `indexImplementation.Dispose()` was called which closes the lucene directory. But we were still using it in ` TryRecoveringIndex`
https://github.com/ravendb/ravendb/blob/8936425e9eda9975a189890cae73d16eaba78b85/Raven.Database/Indexing/IndexStorage.cs#L278